### ISSUE TITLE
Boot faied hang at loop: module loaded 

### ISSUE THREAD

**Issue by langccc** [eugene-tarassov/vivado-risc-v#81]
I reserved some memory like this:
![image](https://user-images.githubusercontent.com/64479628/164258004-72b997ee-7ec2-4e14-baeb-2c23b3fd7d0d.png)

but failed when boot linux:
`ERROR: reserving fdt memory region failed (addr=b0000000 size=10000000 flags=4) `
And boot failed,hang at loop: module loaded ,cam you give me some advice to fix it?
This is the boot information:

`Boot 2022.01-dirty (Apr 15 2022 - 05:07:32 -0700)                             
                                                                                
CPU:   rv64imafdc                                                               
Model: freechips,rocketchip-vivado                                              
DRAM:  1 GiB                                                                    
MMC:   mmc0@60000000: 0                                                         
Loading Environment from nowhere... OK                                          
In:    uart@60010000                                                            
Out:   uart@60010000                                                            
Err:   uart@60010000                                                            
Net:   No ethernet found.                                                       
Hit any key to stop autoboot:  0                                                
switch to partitions #0, OK                                                     
mmc0 is current device                                                          
Scanning mmc 0:1...                                                             
Found /extlinux/extlinux.conf                                                   
Retrieving file: /extlinux/extlinux.conf                                        
1:      Debain v5.16.14                                                         
Retrieving file: /extlinux/initrd-v5.16.14.img                                  
Retrieving file: /extlinux/image-v5.16.14                                       
append: ro root=UUID=68d82fa1-1bb5-435f-a5e3-862176586eec earlycon initramfs.ru8
Moving Image from 0x81000000 to 0x80200000, end=8131a190                        
## Flattened Device Tree blob at 00010080                                       
   Booting using the fdt blob at 0x010080                                       
ERROR: reserving fdt memory region failed (addr=b0000000 size=10000000 flags=4) 
   Loading Device Tree to 0000000084ffc000, end 0000000084ffff40 ... OK         
                                                                                
Starting kernel ...                                                             
                                                                                
[    0.000000] Linux version 5.16.14-dirty (eda@ubuntu) (riscv64-linux-gnu-gcc 2
[    0.000000] OF: fdt: Ignoring memory range 0x80000000 - 0x80200000           
[    0.000000] Machine model: freechips,rocketchip-vivado                       
[    0.000000] earlycon: axi_uart0 at MMIO 0x0000000060010000 (options '')      
[    0.000000] printk: bootconsole [axi_uart0] enabled                          
[    0.000000] Zone ranges:                                                     
[    0.000000]   DMA32    [mem 0x0000000080200000-0x00000000bfffffff]           
[    0.000000]   Normal   empty                                                 
[    0.000000] Movable zone start for each node                                 
[    0.000000] Early memory node ranges                                         
[    0.000000]   node   0: [mem 0x0000000080200000-0x00000000afffffff]          
[    0.000000]   node   0: [mem 0x00000000b0000000-0x00000000bfffffff]          
[    0.000000] Initmem setup node 0 [mem 0x0000000080200000-0x00000000bfffffff] 
[    0.000000] On node 0, zone DMA32: 512 pages in unavailable ranges           
[    0.000000] SBI specification v0.3 detected                                  
[    0.000000] SBI implementation ID=0x1 Version=0x10000                        
[    0.000000] SBI TIME extension detected                                      
[    0.000000] SBI IPI extension detected                                       
[    0.000000] SBI RFENCE extension detected                                    
[    0.000000] SBI v0.2 HSM extension detected                                  
[    0.000000] riscv: ISA extensions acdfim                                     
[    0.000000] riscv: ELF capabilities acdfim                                   
[    0.000000] percpu: Embedded 16 pages/cpu s27544 r8192 d29800 u65536         
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 258055    
[    0.000000] Kernel command line: ro root=UUID=68d82fa1-1bb5-435f-a5e3-8621768
[    0.000000] Dentry cache hash table entries: 131072 (order: 8, 1048576 bytes)
[    0.000000] Inode-cache hash table entries: 65536 (order: 7, 524288 bytes, l)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off          
[    0.000000] Memory: 729772K/1046528K available (6183K kernel code, 4837K rwd)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=2, Nodes=1       
[    0.000000] rcu: Hierarchical RCU implementation.                            
[    0.000000] rcu:     RCU restricting CPUs from NR_CPUS=32 to nr_cpu_ids=2.   
[    0.000000]  Tracing variant of Tasks RCU enabled.                           
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 25 ji.
[    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=2     
[    0.000000] NR_IRQS: 64, nr_irqs: 64, preallocated irqs: 0                   
[    0.000000] riscv-intc: 64 local interrupts mapped                           
[    0.000000] plic: interrupt-controller@c000000: mapped 8 interrupts with 2 h.
[    0.000000] random: get_random_bytes called from start_kernel+0x44e/0x604 wi0
[    0.000000] riscv_timer_init_dt: Registering clocksource cpuid [0] hartid [0]
[    0.000000] clocksource: riscv_clocksource: mask: 0xffffffffffffffff max_cycs
[    0.000021] sched_clock: 64 bits at 800kHz, resolution 1250ns, wraps every 4s
[    0.010395] Calibrating delay loop (skipped), value calculated using timer f)
[    0.021567] pid_max: default: 32768 minimum: 301                             
[    0.028620] Mount-cache hash table entries: 2048 (order: 2, 16384 bytes, lin)
[    0.036986] Mountpoint-cache hash table entries: 2048 (order: 2, 16384 bytes)
[    0.072150] ASID allocator disabled (0 bits)                                 
[    0.078987] rcu: Hierarchical SRCU implementation.                           
[    0.091347] smp: Bringing up secondary CPUs ...                              
[    0.108396] smp: Brought up 1 node, 2 CPUs                                   
[    0.120911] devtmpfs: initialized                                            
[    0.152938] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, ms
[    0.163768] futex hash table entries: 512 (order: 3, 32768 bytes, linear)    
[    0.178173] NET: Registered PF_NETLINK/PF_ROUTE protocol family              
[    0.192996] thermal_sys: Registered thermal governor 'step_wise'             
[    0.193576] thermal_sys: Registered thermal governor 'user_space'            
[    0.380521] vgaarb: loaded                                                   
[    0.394301] SCSI subsystem initialized                                       
[    0.405588] usbcore: registered new interface driver usbfs                   
[    0.412552] usbcore: registered new interface driver hub                     
[    0.419066] usbcore: registered new device driver usb                        
[    0.439620] clocksource: Switched to clocksource riscv_clocksource           
[    0.477107] NET: Registered PF_INET protocol family                          
[    0.486218] IP idents hash table entries: 16384 (order: 5, 131072 bytes, lin)
[    0.520731] tcp_listen_portaddr_hash hash table entries: 512 (order: 1, 8192)
[    0.530601] TCP established hash table entries: 8192 (order: 4, 65536 bytes,)
[    0.541898] TCP bind hash table entries: 8192 (order: 5, 131072 bytes, linea)
[    0.553371] TCP: Hash tables configured (established 8192 bind 8192)         
[    0.563210] UDP hash table entries: 512 (order: 2, 16384 bytes, linear)      
[    0.572210] UDP-Lite hash table entries: 512 (order: 2, 16384 bytes, linear) 
[    0.582293] NET: Registered PF_UNIX/PF_LOCAL protocol family                 
[    0.597681] RPC: Registered named UNIX socket transport module.              
[    0.604440] RPC: Registered udp transport module.                            
[    0.609860] RPC: Registered tcp transport module.                            
[    0.615183] RPC: Registered tcp NFSv4.1 backchannel transport module.        
[    0.622750] PCI: CLS 0 bytes, default 64                                     
[    0.633645] Unpacking initramfs...                                           
[    0.645240] workingset: timestamp_bits=62 max_order=18 bucket_order=0        
[    0.758993] squashfs: version 4.0 (2009/01/31) Phillip Lougher               
[    0.781903] NFS: Registering the id_resolver key type                        
[    0.788011] Key type id_resolver registered                                  
[    0.792993] Key type id_legacy registered                                    
[    0.799011] nfs4filelayout_init: NFSv4 File Layout Driver Registering...     
[    0.806803] nfs4flexfilelayout_init: NFSv4 Flexfile Layout Driver Registerin.
[    0.818711] 9p: Installing v9fs 9p2000 file system support                   
[    0.830102] NET: Registered PF_ALG protocol family                           
[    0.837736] Block layer SCSI generic (bsg) driver version 0.4 loaded (major )
[    0.845940] io scheduler mq-deadline registered                              
[    0.851142] io scheduler kyber registered                                    
[    1.954487] Serial: 8250/16550 driver, 4 ports, IRQ sharing disabled         
[    1.980811] 60010000.uart: ttyAU0 at MMIO 0x60010000 (irq = 1, base_baud = 0t
[    1.990312] printk: console [ttyAU0] enabled                                 
[    1.990312] printk: console [ttyAU0] enabled                                 
[    1.999627] printk: bootconsole [axi_uart0] disabled                         
[    1.999627] printk: bootconsole [axi_uart0] disabled                         
[    2.138301] loop: module loaded
`

**Comment by Rucadi** [comment]
Maybe pmem is useful for your requirements?
Just make your ram smaller and add the pmem on the device-tree.

```
pmem@b0000000  {
		compatible = "pmem-region";
		reg = <0xb0000000 0x10000000>;
		volatile;

};
```

**Comment by flyxia624** [comment]
> 

Thank you ,I will try it,and I want to know why reserved-memory failed，Do you know something about this?

**Comment by langccc** [comment]
I removed reserved memory,and I got a strange error and boot failed:
`U-Boot 2022.01-dirty (Apr 15 2022 - 05:07:32 -0700)                                                     
                                                                                                        
CPU:   rv64imafdc                                                                                       
Model: freechips,rocketchip-vivado                                                                      
DRAM:  1 GiB                                                                                            
MMC:   mmc0@60000000: 0                                                                                 
Loading Environment from nowhere... OK                                                                  
In:    uart@60010000                                                                                    
Out:   uart@60010000                                                                                    
Err:   uart@60010000                                                                                    
Net:   No ethernet found.                                                                               
Hit any key to stop autoboot:  0                                                                        
switch to partitions #0, OK                                                                             
mmc0 is current device                                                                                  
Scanning mmc 0:1...                                                                                     
Found /extlinux/extlinux.conf                                                                           
Retrieving file: /extlinux/extlinux.conf                                                                
1:      Debain v5.16.14                                                                                 
Retrieving file: /extlinux/initrd-v5.16.14.img                                                          
Error reading cluster                                                                                   
** Unable to read file /extlinux/initrd-v5.16.14.img **                                                 
Skipping Debain v5.16.14 for failure retrieving initrd                                                  
SCRIPT FAILED: continuing...                                                                            
** Unable to read file / **                                                                             
Failed to load '/'                                                                                      
libfdt fdt_check_header(): FDT_ERR_BADMAGIC                                                             
Scanning disk mmc0@60000000.blk...                                                                      
Found 3 disks                                                                                           
No EFI system partition                                                                                 
BootOrder not defined                                                                                   
EFI boot manager: Cannot load any image                                                                 
No ethernet found.                                                                                      
No ethernet found.   `
How could reserved-memory node influence accessing SD card?
